src/librustc/macros.rs

```diff
@@ -388,16 +388,16 @@ macro_rules! EnumTypeFoldableImpl {
         {
             fn super_fold_with<'gcx: $tcx, V: $crate::ty::fold::TypeFolder<'gcx, $tcx>>(
                 &self,
-                folder: &mut V,
+                _folder: &mut V,
             ) -> Self {
-                EnumTypeFoldableImpl!(@FoldVariants(self, folder) input($($variants)*) output())
+                EnumTypeFoldableImpl!(@FoldVariants(self, _folder) input($($variants)*) output())
             }
 
             fn super_visit_with<V: $crate::ty::fold::TypeVisitor<$tcx>>(
                 &self,
-                visitor: &mut V,
+                _visitor: &mut V,
             ) -> bool {
-                EnumTypeFoldableImpl!(@VisitVariants(self, visitor) input($($variants)*) output())
+                EnumTypeFoldableImpl!(@VisitVariants(self, _visitor) input($($variants)*) output())
```

Comment:
If the `visitor` isn't used, maybe you should be using `CloneTypeFoldableAndLiftImpls`?

Reply:
Done.